### PR TITLE
[WIP] #8722 Add support for RDS cluster snapshot tags

### LINKED ISSUE #8722: Add support for RDS cluster snapshot tags
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The `aws_db_cluster_snapshot` resource currently does not support the `tags` argument, although adding tags is supported via the REST API, CLI, and console.

### New or Affected Resource(s)

* aws_db_cluster_snapshot

### Potential Terraform Configuration

```hcl
resource "aws_db_cluster_snapshot" "test" {
  db_cluster_identifier = "some-identifier"
  db_cluster_snapshot_identifier = "my-test-snapshot"

  tags {
    "my:application" = "test"
  }
}
```

### References

* https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBClusterSnapshot.html#API_CreateDBClusterSnapshot_RequestParameters
* https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-cluster-snapshot.html#options

### PR DESCRIPTION
**Pull request by aybuke**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Fixes #8722

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
New Argument: Tags for RDS cluster snapshot
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSDBClusterSnapshot_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSDBClusterSnapshot_ -timeout 120m
=== RUN   TestAccAWSDBClusterSnapshot_basic
=== PAUSE TestAccAWSDBClusterSnapshot_basic
=== CONT  TestAccAWSDBClusterSnapshot_basic
--- PASS: TestAccAWSDBClusterSnapshot_basic (222.96s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	223.010s
...
```

### CODE REVIEW

File: aws/resource_aws_db_cluster_snapshot_test.go [hunk 1]
```diff
@@ -39,6 +40,8 @@ func TestAccAWSDBClusterSnapshot_basic(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "status", "available"),
 					resource.TestCheckResourceAttr(resourceName, "storage_encrypted", "false"),
 					resource.TestMatchResourceAttr(resourceName, "vpc_id", regexp.MustCompile(`^vpc-.+`)),
+					testAccCheckDbClusterSnapshotExists("aws_db_cluster_snapshot.test", &v),
```
Comment (bflad) [inline aws/resource_aws_db_cluster_snapshot_test.go]: This check appears to be a duplicate of the existing one on line 28.

File: vendor/github.com/aws/aws-sdk-go/service/rds/api.go [hunk 2]
```diff
@@ -21378,6 +21378,7 @@ type DescribeDBClusterSnapshotsInput struct {
 	// when SnapshotType is set to shared. The IncludeShared parameter doesn't apply
 	// when SnapshotType is set to public.
 	SnapshotType *string `type:"string"`
+	Tags []*Tag `locationNameList:"Tag" type:"list"`
```
Comment (bflad) [inline vendor/github.com/aws/aws-sdk-go/service/rds/api.go]: Hmm. My guess is that if modifying the AWS Go SDK like this is working, this is likely a feature to be added in an upcoming AWS Go SDK update when the RDS service team officially publishes pending API model changes after a production rollout completes. Unfortunately though, please note that we cannot accept changes that are not officially published in the AWS Go SDK for support reasons and that changes like this to the `vendor` files will get automatically reverted during any future `go mod` operations.

For now please revert this pull request to using the available [`ListTagsForResource`](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#RDS.ListTagsForResource) call and we can get this functionality in. Thanks.

File: aws/resource_aws_db_cluster_snapshot_test.go (hunk 1 above)
Comment (aybuke) [reply aws/resource_aws_db_cluster_snapshot_test.go]: I removed duplicated line.

File: vendor/github.com/aws/aws-sdk-go/service/rds/api.go (hunk 2 above)
Comment (aybuke) [reply vendor/github.com/aws/aws-sdk-go/service/rds/api.go]: I sent new commit about this problem but i'm not sure. maybe i misunderstand you. If i did so, i will try again fix this and thank you for your time.:)

Comment (bflad) [reply vendor/github.com/aws/aws-sdk-go/service/rds/api.go]: Hi @aybuke 👋 To clarify: we cannot accept changes to just the `vendor` directory contents. Any changes in the `vendor` directory must occur due to running `go mod vendor`.

This pull request needs to be changed to work without this new line here.

**Comment by nywilken** [comment]
Hi @aybuke thanks again for your time in contributing this change. Unfortunately, we can not accept this change for the reasons outlined by @bflad [here](https://github.com/terraform-providers/terraform-provider-aws/pull/8928#discussion_r296230613). I'm going to close this PR, but please feel free to open a new PR without the vendor changes if you would like to continue working on adding tag support for cluster snapshots.

**Pull request closed without merging**